rss/lambda/lambda.py

```diff
@@ -0,0 +1,69 @@
+import os
+import sys
+import time
+import json
+import botocore, boto3
+
+# This is stupid, but you can't explicitly import module dependencies.
+sys.path.append('./libs')
+
+import feedparser
+import requests
+
+
+def handler(event, context):
+    db = boto3.resource(
+        "dynamodb",
+        region_name=os.environ["dynamo_region"],
+        config=botocore.client.Config(),
+    ).Table(os.environ["dynamo_table"])
+
+    response = db.query(
+        KeyConditionExpression="id = :id",
+        ExpressionAttributeValues={":id": "lambda.py"},
+    )
+    if len(response["Items"]) == 0:
+        return { "statusCode": 500, "body": json.dumps(f"database not configured"), }
+
+    firstResponse = response["Items"][0]
+    lastUpdateSeen = float(firstResponse["ts"])
+    sharedSecret = firstResponse["ss"]
+
+    d = feedparser.parse(
+        "https://htcondor-wiki.cs.wisc.edu/index.cgi/timeline.rss?d=90"
+    )
+    lastUpdate = time.mktime(d.feed.updated_parsed)
+
+    if lastUpdateSeen < lastUpdate:
+        db.update_item(
+            Key={"id": "lambda.py"},
+            UpdateExpression="set ts = :ts",
+            ExpressionAttributeValues={":ts": str(lastUpdate)},
+            ReturnValues="UPDATED_NEW",
+        )
```

Comment:
I'm unfamiliar with DynamoDB, so this looked a little crazy to me at first. Could you move all of the database interaction into functions so that the uninitiated (me :) ) know what they do at a glance?

Reply:
It is a little crazy.  There's probably a way to tie (or whatever the Python equivalent is) a DynamoDB entry to a dictionary, but I didn't think to look for one.